cli/src/client/mod.rs

```diff
@@ -1,39 +1,30 @@
+use crate::errors::{ApolloError, ErrorDetails, Fallible};
 use graphql_client::{GraphQLQuery, Response};
 use http::Uri;
 use reqwest::blocking;
 use std::env;
-use std::error::Error;
 
 mod queries;
 
-static PROD_GQL_API_URL: &str = "https://engine-graphql.apollographql.com/api/graphql";
-
-fn api_uri() -> Uri {
-    env::var("APOLLO_API_URL")
-        .ok()
-        .and_then(|url| url.parse::<Uri>().ok())
-        .unwrap_or_else(|| PROD_GQL_API_URL.parse::<Uri>().unwrap())
-}
-
 pub struct Client {
     api_key: String,
     uri: Uri,
     reqwest: blocking::Client,
 }
 
 impl Client {
-    fn from(api_key: String) -> Client {
+    pub(crate) fn from(api_key: String, uri: String) -> Client {
         Client {
             api_key,
-            uri: api_uri(),
+            uri: uri.parse::<Uri>().unwrap(),
```

Comment:
I think we should be avoiding `unwrap` unless we know we 100% can't fail. This can fail if this function is invoked with an invalid `Uri` (e.g. from an env variable, assuming you'll restore functionality), so at the very least, use `expect` with a descriptive message?